Resources/public/js/views/ezsearch-searchlistview.js

```diff
@@ -8,37 +8,96 @@ YUI.add('ezsearch-searchlistview', function (Y) {
     Y.namespace('eZSearch');
 
     Y.eZSearch.SearchListView = Y.Base.create('ezsearchSearchListView', Y.eZ.TemplateBasedView, [], {
-      
+        initializer: function () {
+            this._itemViews = [];
+        },
         render: function () {
             this.get('container').setHTML(this.template({
-                searchResultItems: this._getResultItems(),
                 searchResultCount: this.get('searchResultCount'),
             }));
+            this._renderItems();
             return this;
         },
+
+        /**
+         * Destroys the instantiated item views.
+         *
+         * @method _destroyItemViews
+         * @protected
+         */
+        _destroyItemViews: function () {
+            this._itemViews.forEach(function(view) {
+                view.destroy();
+            });
+            this._itemViews = [];
+        },
         
-        _getResultItems: function () {
-            var searchResultItems = [];
+        _renderItems: function () {
+            var contentNode = this.get('container').one('.ezsearch-searchlist-content'),
+                ItemView = this.get('itemViewConstructor');
 
             if ( !this.get('searchResultList') ) {
-                return this.get('searchResultList');
+                return;
             }
-
+            this._destroyItemViews();
```

Comment:
there's nothing to destroy here but indeed this method should be called from Y.eZSearch.SearchListView `destructor`